### DISCUSSION TITLE
Any way to rename across all files in the current project directory?

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#2646] [Q&A]
In lazyvim, `<leader>cr` can rename a function/variable in the current file.

May I ask if there is a way, via lazyvim, to do rename across all the files in the current project directory, e.g. for a function used in multiple files? 

Thanks.

**Comment by lainegates** [comment]
if you are writing c/c++, try clangd with https://www.lazyvim.org/extras/lang/clangd
for python, try ruff_lsp

**Comment by pheuberger** [comment]
have you tried `<leader>cr` followed by `:wa`? it's working for me. when i'm renaming a function in one file it will change the call sites and imports in other files as well.

**Comment by hopezh** [comment]
I just tried again in my lazyvim fork, and it seems that  using `<leader>cr` DOES change the name of a function both in the files it's used and in the source file where it was defined.

https://github.com/LazyVim/LazyVim/assets/23435810/410e5d02-c749-4295-9451-c632c720146b
